src/player.cpp

```diff
@@ -6776,6 +6776,10 @@ void player::check_and_recover_morale()
     }
 }
 
+void player::on_worn_item_transform( const item &it ) {
+	morale->on_worn_item_transform( it );
```

Comment:
Is it not sufficient to just call `update_squeamish_penalty` here instead of creating a dummy item and quickly cycling it on the player? I think the recalculation there would be sufficient.

Reply:
Nope, take a look at what happens in `set_worn()`.  It's also better practice to put the code handling the details of morale in `player_morale` not `player`.